src/LayoutManager/LayoutManager.ts

```diff
@@ -0,0 +1,294 @@
+  }
+
+  /**
+   * subscribe to object layout triggers
+   */
+  protected subscribe(context: StrictLayoutContext, object: FabricObject) {
+    const { target } = context;
+    this.unsubscribe(context, object);
+    const disposers = [
+      object.on('modified', (e) =>
+        this.performLayout({
+          trigger: 'modified',
+          e: { ...e, target: object },
+          type: 'object_modified',
+          target,
+        })
+      ),
+      ...(
+        [
+          'moving',
+          'resizing',
+          'rotating',
+          'scaling',
+          'skewing',
+          'changed',
+        ] as TModificationEvents[]
+      ).map((key) =>
+        object.on(key, (e) =>
+          this.performLayout({
+            trigger: key,
+            e: { ...e, target: object },
+            type: 'object_modifying',
+            target,
+          })
+        )
+      ),
+    ];
+    this._subscriptions.set(object, disposers);
+  }
+
+  /**
+   * unsubscribe object layout triggers
+   */
+  protected unsubscribe(context: StrictLayoutContext, object: FabricObject) {
+    (this._subscriptions.get(object) || []).forEach((d) => d());
+    this._subscriptions.delete(object);
+  }
+
+  protected onBeforeLayout(context: StrictLayoutContext) {
+    const { target } = context;
+
+    // handle layout triggers subscription
+    if (context.type === 'initialization' || context.type === 'added') {
+      context.targets.forEach((object) => this.subscribe(context, object));
+    } else if (context.type === 'removed') {
+      context.targets.forEach((object) => this.unsubscribe(context, object));
+    }
+
+    //  fire layout hook and event (event will fire only for layouts after initialization layout)
+    target.onBeforeLayout({
+      context,
+    });
+    target.fire('layout:before', {
+      context,
+    });
+
+    if (context.type === 'imperative' && context.deep) {
+      const { strategy: _, ...tricklingContext } = context;
+      // traverse the tree
+      target.forEachObject((object) => {
+        (object as Group).layoutManager?.performLayout({
+          ...tricklingContext,
+          bubbles: false,
+          target: object as Group,
+        });
+      });
+    }
+  }
+
+  protected getLayoutResult(
+    context: StrictLayoutContext
+  ): Required<LayoutResult> | undefined {
+    const { target } = context;
+    const prevCenter =
+      context.type === 'initialization'
+        ? new Point()
+        : target.getRelativeCenterPoint();
+    const result = context.strategy.calcLayoutResult(
+      context,
+      target.getObjects()
+    );
+    if (!result) {
+      return;
+    }
+    const nextCenter = new Point(result.centerX, result.centerY);
+    const correction = new Point(
+      result.correctionX ?? 0,
+      result.correctionY ?? 0
+    );
+    const relativeCorrection = new Point(
+      result.relativeCorrectionX ?? 0,
+      result.relativeCorrectionY ?? 0
+    );
```

Comment:
this is verbose for no reason.
calcLayoutResult should return center, correction, relativeCorrection as points and we should use them right away instead of recreating them.

Reply:
true